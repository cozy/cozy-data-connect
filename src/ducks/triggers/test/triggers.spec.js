/* eslint-env jest */

import { buildKonnectorTrigger } from '../'

describe('Trigger Duck', () => {
  describe('buildKonnectorTrigger', () => {
    const konnector = { slug: 'test' }
    const account = { _id: '963a51f6cdd34401b0904de32cc5578d' }
    const folder = { _id: 'daa147092e1c4a1da8c991cb2a194adc' }

    const options = {
      frequency: 'weekly',
      day: 1,
      hours: 14,
      minutes: 15
    }

    it('creates a trigger', () => {
      expect(
        buildKonnectorTrigger(konnector, account, folder, options)
      ).toMatchSnapshot()
    })

    it('creates a trigger without folder', () => {
      expect(
        buildKonnectorTrigger(konnector, account, null, options)
      ).toMatchSnapshot()
    })
  })
})
