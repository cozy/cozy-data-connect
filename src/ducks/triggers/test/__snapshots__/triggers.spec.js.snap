exports[`Trigger Duck buildKonnectorTrigger creates a trigger 1`] = `
Object {
  "_type": "io.cozy.triggers",
  "attributes": Object {
    "arguments": "0 15 14 * * 1",
    "type": "@cron",
    "worker": "konnector",
    "worker_arguments": Object {
      "account": "963a51f6cdd34401b0904de32cc5578d",
      "folder_to_save": "daa147092e1c4a1da8c991cb2a194adc",
      "konnector": "test",
    },
  },
}
`;

exports[`Trigger Duck buildKonnectorTrigger creates a trigger without folder 1`] = `
Object {
  "_type": "io.cozy.triggers",
  "attributes": Object {
    "arguments": "0 15 14 * * 1",
    "type": "@cron",
    "worker": "konnector",
    "worker_arguments": Object {
      "account": "963a51f6cdd34401b0904de32cc5578d",
      "konnector": "test",
    },
  },
}
`;
